src/pbft/pbft_config.h

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#include "libbyz/libbyz.h"
+#include "libbyz/pbft_assert.h"
+#include "pbft/pbft_deps.h"
+
+namespace pbft
+{
+  class AbstractPBFTConfig
+  {
+  public:
+    static char* service_mem;
+    virtual ~AbstractPBFTConfig() = default;
+    virtual void set_service_mem(char* sm) = 0;
+    virtual ExecCommand get_exec_command() = 0;
+    virtual size_t message_size() = 0;
+    virtual void fill_request(
+      uint8_t* buffer,
+      size_t total_req_size,
+      const std::vector<uint8_t>& data,
+      size_t jsonrpc_id) = 0;
+  };
+
+  char* AbstractPBFTConfig::service_mem = 0;
+
+  class PbftConfigCCF : public AbstractPBFTConfig
```

Comment:
Minor: Should we stick to PBFT or pbft? I'm happy either way as long as it is consistent.

Reply:
I thought about it a bit too but then we also have the PbftEnclaveNetwork class and the plain Pbft one which might look a bit off as PBFT since we also use that as a defined variable, so I am leaning towards Pbft vs PBFT.

Reply:
We also have `RpcFrontend`, `JsonField`, etc, so I think `Pbft` matches our coding style.

Reply:
I'd have prefered all caps, they are acronyms, not words. I'd rather stay consistent and minimise change though, so I agree with @eddyashton.

Reply:
Apparently the Microsoft-style is to [CamelCase any acronyms or abbreviations over 2 letters long](https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/141e06ef(v=vs.71)).

To get right into the weeds: this should really be `PbftConfigCcf`, rather than `CCF`.

Reply:
Changing that too to be consistent. I don't know that we're following the .NET style particularly, if we're going to following naming conventions I'd rather go with the ISO Cpp Core guidelines: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-name-type